### PR TITLE
Refactored environment variables using envparse

### LINKED ISSUE #150: ENH: use envparse to parse environment variable values
**Is it related to a problem?**
It solves #149, but it's not limited to that. It is a more general refactoring that can prevent other similar bugs.

**API breaking implications**
None

### PR DESCRIPTION
**Pull request by MJafarMashhadi**
closes #150 
xref #151 , #150

### CODE REVIEW

File: train/gs_url.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import os
+from envparse import env
 from pathlib import Path
 
 
 def build_raw_data_dir() -> str:
-    bucket = os.environ.get('GOOGLE_AI_PLATFORM_BUCKET')
-    assert bucket, "GCS bucket not defined"
+    bucket = env('GOOGLE_AI_PLATFORM_BUCKET', default=None)
+    assert bucket, "GCS bucket not defined"  # TODO: If no default is set env will automatically raise an exception, so we might be able to remove this assertion
```
Comment (jzhang-gp) [inline train/gs_url.py:7]: If the default value is None, does it still throw exceptions?

File: train/gs_url.py [hunk 2]
```diff
@@ -14,8 +14,8 @@ def build_raw_data_url(dataset_name) -> str:
 
 
 def build_model_dir(model_uuid, model_version) -> str:
-    bucket = os.environ.get('GOOGLE_AI_PLATFORM_BUCKET')
-    assert bucket, "GCS bucket not defined"
+    bucket = env('GOOGLE_AI_PLATFORM_BUCKET', default=None)
+    assert bucket, "GCS bucket not defined"  # TODO: above
```
Comment (jzhang-gp) [inline train/gs_url.py:18]: Above meaning the throwing exception comment you have above? Right?

**Review by jzhang-gp — COMMENTED** [review]
It seems that this PR already covers the other one. Have you rebuilt the images and verified the website is still working?

File: train/gs_url.py (hunk 1 above)
Comment (MJafarMashhadi) [reply train/gs_url.py:7]: No, so `env('NON_EXISTENT')` will raise an exception, `env('NON_EXISTENT', default=None)` will just return None.

File: train/gs_url.py (hunk 2 above)
Comment (MJafarMashhadi) [reply train/gs_url.py:18]: Correct

**Review by jzhang-gp — APPROVED** [review]
If the change has been tested in the new training image, feel free to merge the change.

**Comment by MJafarMashhadi** [comment]
Thanks, the tests pass. The training image works too.

**Pull request merged**